### PR TITLE
Replace all pages to redirect

### LINKED ISSUE #3899: Disable or update https://xonsh.github.io/
Hi! xonsh is the one small step for a snail and one giant leap for mankind!

Please make the redirect from https://xonsh.github.io/ to https://xon.sh/contents.html.

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### PR DESCRIPTION
**Pull request by anki-code**
1. Replaced all pages to redirect
2. Added `robots.txt` with disabling search index

Closes https://github.com/xonsh/xonsh/issues/3899

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,429 +1 @@
-
```
Comment (scopatz) [inline index.html:1]: Doesn't index need the redirect?

Comment (anki-code) [reply index.html:1]: Why?

Comment (anki-code) [reply index.html:1]: Or you mean that it should be to xon.sh?

Comment (anki-code) [reply index.html:1]: Changed: index.html -> xon.sh

Comment (anki-code) [reply index.html:1]: It's not needed to set the detailed redirects because after time the old pages will go from search index.

Comment (scopatz) [reply index.html:1]: It would still be good in case people come to this site directly.

Comment (anki-code) [reply index.html:1]: Don't clearly understand what you want.

Comment (scopatz) [reply index.html:1]: Just a redirect like the other pages have

Comment (anki-code) [reply index.html:1]: The `index.html` will be `<html><script>window.location.replace("https://xon.sh/");</script><body>Redirect to https://xon.sh/</body></html>` - like the other html pages. What you mean? )

Comment (scopatz) [reply index.html:1]: Yeah that is what I mean

Comment (anki-code) [reply index.html:1]: It's already as you want ) Probably Github just hidden the info.

Comment (scopatz) [reply index.html:1]: oh you are right, thanks

**Pull request merged**